### PR TITLE
Remove the default PCI host devices

### PR DESCRIPTION
**Pull request by nunnatsa**
This PR fixes the bug in https://bugzilla.redhat.com/show_bug.cgi?id=1969912

HCO will not add PCI devices by default, letting the system administrator decide if to add them.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Remove the default PCI host devices
```

### CODE REVIEW

File: hack/check_defaults.sh [hunk 1]
```diff
@@ -111,55 +103,6 @@ for JPATH in "${LMPATHS[@]}"; do
     sleep 2
 done
 
-echo "Check that permittedHostDevices defaults are behaving as expected"
-
-./hack/retry.sh 10 3 "${KUBECTL_BINARY} patch hco -n \"${INSTALLED_NAMESPACE}\" --type=json kubevirt-hyperconverged -p '[{ \"op\": \"replace\", \"path\": /spec, \"value\": {} }]'"
-for JPATH in "${PERMITTED_HOST_DEVICES_PATHS[@]}"; do
-    ./hack/retry.sh 10 3 "${KUBECTL_BINARY} patch hco -n \"${INSTALLED_NAMESPACE}\" --type='json' kubevirt-hyperconverged -p '[{ \"op\": \"remove\", \"path\": '\"${JPATH}\"' }]'"
-    PHD=$(${KUBECTL_BINARY} get hco -n "${INSTALLED_NAMESPACE}" kubevirt-hyperconverged -o jsonpath='{.spec.permittedHostDevices}')
-    if ! echo "${PHD}" | grep "${PERMITTED_HOST_DEVICES_DEFAULT1}"; then
```
Comment (tiraboschi) [inline hack/check_defaults.sh:120]: I think we should keep the test logic but removing PERMITTED_HOST_DEVICES_DEFAULT1 and PERMITTED_HOST_DEVICES_DEFAULT2 on line 27, 28 ensuring we get empty defaults.

**Review by tiraboschi — APPROVED** [review]

File: hack/check_defaults.sh (hunk 1 above)
Comment (nunnatsa) [reply hack/check_defaults.sh:120]: But there are no default devices, so there is nothing in the CR.

**Comment by nunnatsa** [comment]
/hold

waiting for the bug to be ack

**Comment by nunnatsa** [comment]
Closing ad re-open to trigger the openshiftci test.

**Review by tiraboschi — APPROVED** [review]

**Review by tiraboschi — APPROVED** [review]

**Comment by tiraboschi** [comment]
ignoring slightly decreased coverage
/override coverage/coveralls

**Review by tiraboschi — APPROVED** [review]

**Comment by tiraboschi** [comment]
ignoring slightly decreased coverage
/override coverage/coveralls

**Review by tiraboschi — APPROVED** [review]

**Comment by tiraboschi** [comment]
hco-e2e-upgrade-index-aws and hco-e2e-upgrade-index-azure are not supposed to pass due to the disappear of defaulter-hco.kubevirt.io.
/override ci/prow/hco-e2e-upgrade-index-aws
/override ci/prow/hco-e2e-upgrade-index-azure

**Comment by tiraboschi** [comment]
there is nothing specific that should fail just on OKD and not on OCP.
Everything here looks just like infra issues.

**Comment by tiraboschi** [comment]
On https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kubevirt_hyperconverged-cluster-operator/1384/pull-ci-kubevirt-hyperconverged-cluster-operator-main-okd-hco-e2e-image-index-aws/1403327902874865664 I still see:

```
[*6mINFO[*0*1-06-11T14:04:**Z] + grep '{"pciDeviceSelector":"10DE:1DB6","resourceName":"nvidia.com/GV100GL_Tesla_V100"}' 
[*6mINFO[*0*1-06-11T14:04:**Z] Failed checking CR defaults for permittedHostDevices 
[*6mINFO[*0*1-06-11T14:04:**Z] + echo 'Failed checking CR defaults for permittedHostDevices' 
[*6mINFO[*0*1-06-11T14:04:**Z] + exit 1                                     
[*6mINFO[*0*1-06-11T14:04:**Z] Exiting... Last phase status: Failed 
```

I fear that we have some configuration issue with the OKD lanes

**Comment by tiraboschi** [comment]
Ignoring OKD lanes due to infra issues.
/override ci/prow/okd-hco-e2e-image-index-aws
/override ci/prow/okd-hco-e2e-image-index-azure
/override ci/prow/okd-hco-e2e-upgrade-index-azure 
/override ci/prow/okd-hco-e2e-upgrade-index-aws

**Pull request merged**